packages/plugins/foundry-contracts/ChugSplash.sol

```diff
@@ -58,17 +73,76 @@ contract ChugSplash is Script, Test {
         }
     }
 
-    function _initializeChugSplash() private {
-        string[] memory cmds = new string[](7);
+    function getBootloaderBytecode() private returns (DeploymentBytecode memory) {
+        string[] memory cmds = new string[](4);
         cmds[0] = "npx";
         cmds[1] = "node";
         cmds[2] = filePath;
-        cmds[3] = "initializeChugSplash";
-        cmds[4] = rpcUrl;
-        cmds[5] = network;
-        cmds[6] = privateKey;
+        cmds[3] = "getBootloaderBytecode";
+
+        bytes memory result = vm.ffi(cmds);
+        return abi.decode(result, (DeploymentBytecode));
+    }
 
-        vm.ffi(cmds);
+    function _ensureChugSplashInitialized() private {
+        // Fetch bytecode from artifacts
+        DeploymentBytecode memory bootloaderBytecode = getBootloaderBytecode();
+
+        // If the registry is not already deployed
+        if (bootloaderBytecode.registryAddress.code.length == 0) {
+            // If the target chain is a local network
+            if (block.chainid == 31337) {
```

Comment:
Hard-coding `31337` won't work if the user spins up a local node with a different chain ID. I looked through the Foundry docs pretty thoroughly, and couldn't think of a way to determine if the current network is local or not. I think the best option is FFI.

Reply:
So just using ffi won't work for the internal foundry node, but we could probably do something where we first check if the chain id is `31337`. Then if there is an rpc url provided, we could use ffi to check if that node is anvil node using our `isLiveNetwork` function.

Reply:
That unfortunately won't work if the user changes the chain ID of the internal foundry node (i.e. `forge script --chain-id <number>`). Also, I'm not the biggest fan of our solution to get the current rpc url, since it requires the user to hard-code a `NETWORK` var in their `.env` file, which may be incorrect. I wasn't able to figure out a better solution yet, but it's probably possible to do better